### ISSUE TITLE
expand tsk_table_collection_equals?

### ISSUE THREAD

**Issue by mufernando** [tskit-dev/tskit#731]
I don't know how useful this could be to other people, but I always find myself trying to compare two TableCollections. There is a method to check if two table collections are the same, but it would be useful to report which tables differ.

**Comment by mufernando** [comment]
I was thinking a `tsk_bookmark_t` with trues and falses.

**Comment by jeromekelleher** [comment]
Good idea. We'd need a new function though, we don't want to change the API for ``tsk_table_collection_equals``.
